app/controllers/todo_lists_controller.rb

```diff
@@ -1,12 +1,19 @@
 class TodoListsController < ApplicationController
+
   before_action :authenticate_user!   # Ensure the user is authenticated via token
   before_action :set_user
   before_action :set_todo_list, only: [:show, :update, :destroy]
+  
 
   # GET /users/:user_id/todo_lists
   def index
-    @todo_lists = @user.todo_lists
-    render json: @todo_lists
+    @todo_lists = policy_scope(TodoList)
+    if @todo_lists.none?
+      render json: { error: 'Not Found' }, status: :not_found
+    else
+      render json: @todo_lists
+    end  
+    
   end
 
   # POST /users/:user_id/todo_lists
```

Comment:
while creating do we need authorization

Reply:
yes there should be authorization required  as to ensure only authorized users can perform the action.